src/GafferScene/LightToCamera.cpp

```diff
@@ -0,0 +1,312 @@
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "IECore/CompoundData.h"
+#include "IECore/Camera.h"
+#include "IECore/Shader.h"
+#include "IECore/Transform.h"
+#include "OpenEXR/ImathMatrix.h"
+#include "OpenEXR/ImathVec.h"
+#include "GafferScene/LightToCamera.h"
+#include "Gaffer/Metadata.h"
+#include "boost/algorithm/string.hpp"
+
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferScene;
+using namespace Imath;
+
+namespace 
+{
+
+	// TODO: copied from src/GafferSceneUI/StandardLightVisualiser.cpp
+	// Should this live somewhere central? I think John talked about using the same functionality
+	// to drive light manipulators? 
+	template<typename T>
+	T parameter( InternedString metadataTarget, const IECore::CompoundData *parameters, InternedString parameterNameMetadata, T defaultValue )
+	{
+		ConstStringDataPtr parameterName = Metadata::value<StringData>( metadataTarget, parameterNameMetadata );
+		if( !parameterName )
+		{
+			return defaultValue;
+		}
+
+		typedef IECore::TypedData<T> DataType;
+		if( const DataType *parameterData = parameters->member<DataType>( parameterName->readable() ) )
+		{
+			return parameterData->readable();
+		}
+
+		return defaultValue;
+	}
+
+	// If the attributes contain a light, return a corresponding camera
+	// Just returns the first light it finds, which could produce inconsistent results if you
```

Comment:
I don't mind this as a strategy for now, but I don't really buy this as a justification. If someone were making  a multi-renderer setup without the fancy adaptors we have, they might very much want to have light shaders for more than one renderer assigned at the same location.